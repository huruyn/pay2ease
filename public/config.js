window.API_URL = "https://zippy-retail-api-dev.azurewebsites.net";