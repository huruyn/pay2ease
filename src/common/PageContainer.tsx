import PropTypes from 'prop-types';

function PageContainer({
  className,
  title,
  subTitle,
  backdropImage,
  children,
}: {
  className?: string;
  title?: string;
  subTitle?: string;
  backdropImage?: string;
  children: any;
}) {
  return (
    <>
      <div className={className}>
        <div className="zippy-cash-container-wrap">
          <div className={`backdrop ${backdropImage || ''}`} />
          <div className="zippy-cash-container">
            {title && subTitle
              && (
                <div className="zippy-cash-container-header">
                  <h1 className="title">{title}</h1>
                  <h2 className="catch">{subTitle}</h2>
                </div>
              )}
            <div className="zippy-cash-container-body">
              {children}
            </div>
          </div>
        </div>
      </div>
    </>
  );
}

PageContainer.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string,
  subTitle: PropTypes.string,
  backdropImage: PropTypes.string,
};

PageContainer.defaultProps = {
  className: '',
  title: '',
  subTitle: '',
  backdropImage: '',
};

export default PageContainer;
