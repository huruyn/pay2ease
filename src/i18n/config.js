import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

i18n.use(initReactI18next).init({
  fallbackLng: 'en-US',
  lng: 'en-US',
  resources: {
    'en-CA': {
      translations: require('./locales/en-CA/translations.json')
    },
    'en-US': {
      translations: require('./locales/en-US/translations.json')
    },
    'es-US': {
      translations: require('./locales/es-US/translations.json')
    },
    'fr-CA': {
      translations: require('./locales/fr-CA/translations.json')
    },
  },
  ns: ['translations'],
  defaultNS: 'translations'
});

i18n.languages = ['en-CA', 'en-US', 'es-US', 'fr-CA'];

export default i18n;