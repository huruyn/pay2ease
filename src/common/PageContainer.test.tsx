import '../i18n/config';
import PageContainer from './PageContainer';

const ReactTestRenderer = require('react-test-renderer');

describe('PageContainer Component', () => {
  it('matches the snapshot', () => {
    const tree = ReactTestRenderer.create(
      <PageContainer title="sample">
        <div>test</div>
      </PageContainer>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
