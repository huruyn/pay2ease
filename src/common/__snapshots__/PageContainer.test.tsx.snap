// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer Component matches the snapshot 1`] = `
<div
  className=""
>
  <div
    className="zippy-cash-container-wrap"
  >
    <div
      className="backdrop "
    />
    <div
      className="zippy-cash-container"
    >
      
      <div
        className="zippy-cash-container-body"
      >
        <div>
          test
        </div>
      </div>
    </div>
  </div>
</div>
`;
